com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -2166,6 +1971,22 @@ public Map<String, String> getOptions() {
         return options;
     }
 
+    /**
+     * Get the project build state
+     * @return The project build state
+     */
+    public State getState() {
+        return state;
+    }
+
+    /**
+     * Get the resource cache for this project
+     * @return The project resource cache
+     */
+    public ResourceCache getResourceCache() {
+        return resourceCache;
+    }
+
```

Comment:
Please take a look `createGeneratedResource` function  (line 2030) we call it from build steps and it's not `synchronized`, I'm not 100% sure, but maybe it's the reason of issues I'm currently fighting with